java/src/Ice/src/main/java/com/zeroc/Ice/ConnectionI.java

```diff
@@ -2286,6 +2294,14 @@ private boolean sendResponseImpl(OutputStream outputStream, boolean isTwoWay, by
             if (isTwoWay) {
               sendMessage(new OutgoingMessage(outputStream, compress != 0, true));
             }
+
+            if (_state == StateActive && _maxDispatches > 0 && _dispatchCount == _maxDispatches) {
+              // Resume reading if the connection is active and the dispatch count is about to be
+              // less than _maxDispatches.
+              _threadPool.update(this, SocketOperation.None, SocketOperation.Read);
+              _idleTimeoutTransceiver.enableIdleCheck();
```

Comment:
I don't see why we don't need the null check here.

Reply:
Added null check.